lib/diplomat/role.rb

```diff
@@ -0,0 +1,117 @@
+module Diplomat
+  # Methods for interacting with the Consul ACL Role API endpoint
+  class Role < Diplomat::RestClient
+    @access_methods = %i[list read create delete update]
+    attr_reader :id, :type, :acl
+
+    # Read ACL role with the given UUID or name
+    # @param id [String] UUID or name of the ACL role to read
+    # @param options [Hash] options parameter hash
+    # @return [Hash] existing ACL role
+    # rubocop:disable PerceivedComplexity
+    def read(id, options = {}, not_found = :reject, found = :return)
+      endpoint = if id =~ /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i
```

Comment:
I am not a big fan of this too clever trick: I would rather either create 2 functions or allow to provide either id:value or name:value.
What happens for people using another tool generated UUID for the names of roles?

Reply:
I'd say they're doing it wrong 🧀  

But I'll add an extra method `read_name`.

Reply:
Method added with specs.